cardano-node/src/Cardano/Node/Configuration/Socket.hs

```diff
@@ -117,47 +122,39 @@ gatherConfiguredSockets NodeConfiguration { ncNodeIPv4Addr,
     systemDSockets <- liftIO getSystemdSockets
 
     -- Select the sockets or address for public node-to-node comms
-    --
-    let --TODO: add config file support
-        ipv4Sockets, ipv6Sockets :: Maybe [Socket]
-        ipv4Sockets = (\(a, _, _) -> a) <$> systemDSockets
-        ipv6Sockets = (\(_, a, _) -> a) <$> systemDSockets
+    -- TODO: add config file support
+    let -- The first systemd IPv4 socket if it exists
+        firstIpv4Socket :: Maybe Socket
+        firstIpv4Socket = join $ listToMaybe . (\(a, _, _) -> a) <$> systemDSockets
+
+        -- The first systemd IPv6 socket if it exists
+        firstIpv6Socket :: Maybe Socket
+        firstIpv6Socket = join $ listToMaybe . (\(_, a, _) -> a) <$> systemDSockets
```

Comment:
I modified the type from `Maybe [Socket]` to `Maybe Socket` because we never use anything after the first element in the list, so the list is redundant.  This simplification reduces the number of cases that needs handling later, reducing opportunities for errors.